lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -67,3 +75,19 @@ data ErrPostTx
     deriving (Generic, Show, Eq)
 
 instance Exception ErrPostTx
+
+-- | Tries to waits 20 s, until 'networkTip networkLayer' succeeds.
+waitForConnection
+    :: NetworkLayer IO
+    -> IO ()
+waitForConnection nw = loop 20
```

Comment:
can't we reuse `waitForConnection` from `Jourmungadr.Network` module ?

Reply:
Not really. They requests are different* and they are pattern-matching on different errors.

*) Which to me appears like a thing we want: embracing the difference between `NetworkLayer` and the potentially underlying `Jormungandr` client.